cip 0058 Add IntellectEU as a Weight 1 Super Validator (#50)
global-synchronizer-foundation/cips: cip-0058/cip-0058.md

## CIP 0058

<pre>
  CIP: CIP 0058
  Title:  Add IntellectEU as Weight 1 SV
  Author: 
    Christopher Kelly 
  Status: Draft 
  Type: Governance 
  Created: 2025-04-14
  Approved: TBD
  License: CC0-1.0
</pre>

## Abstract

This CIP proposes adding IntellectEU as a Tier 3 Super Validator (SV) to the Canton Network. As part of their role, IntellectEU will deliver infrastructure to support milestone-based rewards, an onchain package manager, and secure custody integrations.

#### Specification
* Deploy a smart contract-based Catalyst Package Manager to manage discovery, versioning, and distribution of Canton applications.
* Implement an onchain escrow mechanism in the GSF multi-tenant SV to hold and release CIP rewards upon milestone completion.

> * Extend the GSF-hosted SV infrastructure to support secure transaction signing through integration with external custody and key management systems (e.g., AWS KMS, Dfns), including support for key rotation, audit logging, and role-based access control. IntellectEU commits to integrating with at least one such provider within the current year.

Reviewer: We don't need these to be in the Specification, because the SVs don't deploy these.  If the SV-executed onchain action depends on these items being completed, we can state that here.

* Register IntellectEU as a Tier 3 SV with weight 1.

## Motivation

The CIP addresses the following challenges in the growing Canton ecosystem:

* Lack of a shared registry for application discovery and updates across participants.
* Inadequate support for enterprise-grade key management and secure transaction signing.
* Absence of accountability mechanisms to ensure Super Validators deliver value in return for rewards.

## Rationale

> * **Canton Package Manager:** A decentralized registry enables network-wide collaboration and efficient deployment. Catalyst’s existing capabilities reduce time-to-delivery and offer a familiar interface to developers, accelerating adoption.

Reviewer: ```suggestion
IntellectEU will make the following solutions available via the Global Synchronizer: 
* **Canton Package Manager:** A decentralized registry enables network-wide collaboration and efficient deployment. Catalyst’s existing capabilities reduce time-to-delivery and offer a familiar interface to developers, accelerating adoption.
```

## CIP 0058

<pre>
  CIP: CIP 0058
  Title:  Add IntellectEU as Weight 1 SV
  Author: 
    Christopher Kelly 
  Status: Approved 
  Type: Governance 
  Created: 2025-04-14
  Approved: 2025-04-25
  License: CC0-1.0
</pre>

## Abstract:

* Add IntellectEU as a Tier3 SV (Weight 1)

> * IntellectEU commits to add a Maintainer to Splice.

Reviewer: ```suggestion
* IntellectEU commits to add a Contributor to Splice.
```

* IntellectEU will meaningfully contribute to the Splice repos at least once per month for at least 1 year, focusing on resolving issues, but without an obligation to complete all listed tasks/issues

> * As the operator of the GSF multi-tenant SV, we will implement the necessary infrastructure in order to operationalize the milestone-based CIP rewards structure as described in the “Mechanics” section within this CIP.

Reviewer: ```suggestion
* As the operator of the GSF multi-tenant SV, IntellectEU will implement the necessary infrastructure in order to operationalize the milestone-based CIP rewards structure as described in the “Mechanics” section within this CIP.
```

**Deliverables for full SV Reward:**

| Deliverable | Acceptance Criteria | Deadline | Weight Earned |
|-------------|---------------------|----------|----------------|
| Enable escrow mechanisms to ensure Super Validators deliver value in return for rewards | As the maintainer of the GSF multi-tenant SV, IntellectEU will implement the necessary infrastructure in order to operationalize the milestone-based CIP rewards structure as described in the “Mechanics” section within this CIP. | +60 days | 0.5 |
| Commit at least one resource to maintain and resolve issues within the Splice repo | IntellectEU will document the resolved issues/contributions to the Splice repo, demonstrating at least one contribution per month. | +90 days | 0.5 |

## Mechanics:

> 

Reviewer: ```suggestion

IntellectEU will implement the following mechanics both for itself, and for all other Super Validators whose SV weights are contigent on completion of specified contributions, following similar mechanics: 
```

**Deliverables for full SV Reward:**

| Deliverable | Acceptance Criteria | Deadline | Weight Earned |
|-------------|---------------------|----------|----------------|
| Enable escrow mechanisms to ensure Super Validators deliver value in return for rewards | As the maintainer of the GSF multi-tenant SV, IntellectEU will implement the necessary infrastructure in order to operationalize the milestone-based CIP rewards structure as described in the “Mechanics” section within this CIP. | +60 days | 0.5 |
| Commit at least one resource to maintain and resolve issues within the Splice repo | IntellectEU will document the resolved issues/contributions to the Splice repo, demonstrating at least one contribution per month. | +90 days | 0.5 |

## Mechanics:

IntellectEU will implement the following mechanics both for itself, and for all other Super Validators whose SV weights are contigent on completion of specified contributions, following similar mechanics: 

> 

Reviewer: ```suggestion
    * ⅔ of the Super Validator Operators will update their configurations to allow GSF to host the full weight to be earned by the given Super Validator
```

> * An ‘escrow’ SV will be setup by the GSF with an SV Weight at the maximum earnable weight in this CIP

Reviewer: ```suggestion
* An `extraBeneficiary` in the name of the ‘escrowed’ Super Validator will be setup by the GSF with an SV Weight at the maximum earnable weight in this CIP
```

> * An ‘escrow’ SV will be setup by the GSF with an SV Weight at the maximum earnable weight in this CIP

Reviewer: ```suggestion
* An `extraBeneficiary` in the name of the ‘escrowed’ Super Validator will be setup by the GSF with an SV Weight at the maximum earnable weight in this CIP
```

> * An `extraBeneficiary` in the name of the ‘escrowed’ Super Validator will be setup by the GSF with an SV Weight at the maximum earnable weight in the CIP that granted rights to that Super Validator.

Reviewer: ```suggestion
* An `extraBeneficiary` PartyID associated with the ‘escrowed’ Super Validator will be setup by the GSF with an SV Weight at the maximum earnable weight in the CIP that granted rights to that Super Validator.
```

    * The Applicant is responsible for all costs associated with the operation of the escrow SV
    * The escrow SV will NOT mint rewards on a block by block basis
    * All escrow SV rewards will go to the Unclaimed Rewards pool
* Applicant is required to present proof of successful completed milestones to the Tokenomics Working Group
    * Applicant is required to present a calculation for number of Canton Coin it should earn for meeting the requirements of the milestone
* If the Tokenomics Working Group agrees the milestone has been met and agrees with the calculation, an announcement will be sent via the Tokenomics-Announce mailing List

>     * ⅔ of Super Validator Operators will then assign a portion of the Unclaimed Rewards to be minted by the Applicant’s Validator

Reviewer: ```suggestion
    * ⅔ of Super Validator Operators will then assign a portion of the Unclaimed Rewards to be minted by the Applicant’s Validator
    * The GSF will update the `extraBeneficiary` to an active PartyID controlled by that Super Validator. 
```

>     * ⅔ of the Super Validator Operators will update their configurations to allow Applicant to takeover a portion of their SV Weight on a go-forward basis

Reviewer: ```suggestion

```

>     * ⅔ of Super Validator Operators will then assign a portion of the Unclaimed Rewards to be minted by the Applicant’s Validator
>     * The GSF will update the `extraBeneficiary` to an active PartyID controlled by that Super Validator. 

Reviewer: ```suggestion
    * The GSF will update the `extraBeneficiary` to an active PartyID controlled by that Super Validator. 
    * ⅔ of Super Validator Operators will then assign a portion of the Unclaimed Rewards to be minted by the Applicant’s Validator, based on the calculation approved by the Tokenomics working group.
   
```

    * The Applicant is responsible for all costs associated with the operation of the escrow SV
    * The escrow SV will NOT mint rewards on a block by block basis
    * All escrow SV rewards will go to the Unclaimed Rewards pool
* Applicant is required to present proof of successful completed milestones to the Tokenomics Working Group
    * Applicant is required to present a calculation for number of Canton Coin it should earn for meeting the requirements of the milestone
* If the Tokenomics Working Group agrees the milestone has been met and agrees with the calculation, an announcement will be sent via the Tokenomics-Announce mailing List
    * The GSF will update the `extraBeneficiary` to an active PartyID controlled by that Super Validator. 
    * ⅔ of Super Validator Operators will then assign a portion of the Unclaimed Rewards to be minted by the Applicant’s Validator, based on the calculation approved by the Tokenomics working group.
   
* If any milestones and associated rewards are not achieved by the deadline
    * Applicant will be notified they have not met a deliverable by the GSF 

>     * Remaining SV Weight on the `extraBeneficiary` SV will be removed from the SV Operator configs, and the total SV weight of the GSF SV node will be reduced by the same amount. 

Reviewer: ```suggestion
    * Remaining SV Weight assigned to the `extraBeneficiary` SV will be removed from the GSF node, and the total SV weight of the GSF SV node will be reduced by the same amount by a vote of the Super Validators.
```
